blogs/cert-manager-tls/yelb-gw.yaml

```diff
@@ -0,0 +1,150 @@
+---
+apiVersion: appmesh.k8s.aws/v1beta2
+kind: VirtualGateway
+metadata:
+  name: yelb-gw
+  namespace: yelb
+spec:
+  backendDefaults:
+    clientPolicy:
+      tls:
+        enforce: true
+        validation:
+          trust:
+            file:
+              certificateChain: /etc/keys/yelb/ca.crt
+  namespaceSelector:
+    matchLabels:
+      gateway: yelb-gw
+  podSelector:
+    matchLabels:
+      app: yelb-gw
+  listeners:
+    - portMapping:
+        port: 8443
+        protocol: http
+      tls:
+        certificate:
+          file:
+            certificateChain: /etc/keys/yelb/tls.crt
+            privateKey: /etc/keys/yelb/tls.key
+        mode: STRICT        
+---
+apiVersion: appmesh.k8s.aws/v1beta2
+kind: GatewayRoute
+metadata:
+  name: gateway-route
+  namespace: yelb
+spec:
+  httpRoute:
+    match:
+      prefix: "/"
+    action:
+      target:
+        virtualService:
+          virtualServiceRef:
+            name: yelb-ui
+---
+apiVersion: cert-manager.io/v1alpha2
+kind: Certificate
+metadata:
+  name: yelb-cert-gw
+  namespace: yelb
+spec:
+  dnsNames:
+    - "yelb-gw.yelb.svc.cluster.local"
+  secretName: yelb-tls-gw
+  issuerRef:
+    name: ca-issuer
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: yelb-gw
+  namespace: yelb
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: yelb-gw
+  template:
+    metadata:
+      labels:
+        app: yelb-gw
+    spec:
+      containers:
+        - name: envoy
+          image: 840364872350.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:v1.15.0.0-prod
+          ports:
+            - containerPort: 8443   
+          volumeMounts:
+           - mountPath: "/etc/keys/yelb"
+             name: yelb-tls-gw
+             readOnly: true
+      volumes:
+        - name: yelb-tls-gw
+          secret:
+            secretName: yelb-tls-gw
+---
+apiVersion: appmesh.k8s.aws/v1beta2
+kind: VirtualNode
+metadata:
+  name: yelb-ui
+  namespace: yelb
+spec:
+  awsName: yelb-ui-virtual-node
+  podSelector:
+    matchLabels:
+      app: yelb-ui
+  listeners:
+    - portMapping:
+        port: 80
+        protocol: http  
+      tls:
+        certificate:
+          file:
+            certificateChain: /etc/keys/yelb/tls.crt
+            privateKey: /etc/keys/yelb/tls.key
+        mode: STRICT         
+  serviceDiscovery:
+    dns:
+      hostname: yelb-ui.yelb.svc.cluster.local
+  backendDefaults:
+    clientPolicy:
+      tls:
+        enforce: true
+        validation:
+          trust:
+            file:
+              certificateChain: /etc/keys/yelb/ca.crt      
+  backends:
+    - virtualService:
+       virtualServiceRef:
+          name: yelb-appserver
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: yelb-gw
+  namespace: yelb
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
+    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: "arn:aws:acm:<region>:123456789012:certificate/a726a8ab-xxxx-xxxx-xxxx-fcb0f86dec6a"
```

Comment:
Given we need to inject the envoy image, I think it makes sense to inject this value using an ENV variable as well.